### PR TITLE
feat: Create Itunes provider for redux store sharing

### PR DESCRIPTION
**Pull request by bhaveshpawar07**
### Ticket Link

---

### Related Links

---

### Description

---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### 
GIF's

iTunesPR6
---

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: app/containers/ITunesProvider/AlbumDetails/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  const { id } = useParams();
+  useEffect(() => {
+    const getFromStore = iTunesData?.results?.find((song) => song.trackId == id);
+
+    if (!getFromStore) {
+      dispatchGetSongDetails(id);
+    } else {
+      songData = {
+        resultCount: 1,
+        results: [getFromStore]
```
Comment (naeemdadi1) [inline app/containers/ITunesProvider/AlbumDetails/index.js]: Why are we storing it in an array?

File: app/containers/ITunesProvider/saga.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  } else {
+    yield put(failureGetItunesData(data));
+  }
+}
+
+export function* getDetails(action) {
+  const response = yield call(getSongDetails, action.songId);
+  const { ok, data } = response;
+  if (ok) {
+    yield put(successGetSongDetails(data));
```
Comment (naeemdadi1) [inline app/containers/ITunesProvider/saga.js]: you can directly pass result to redux store like `data.result[0]` with this you won't have to loop through the data in index file

File: app/containers/ITunesProvider/AlbumDetails/index.js (hunk 1 above)
Comment (bhaveshpawar07) [reply app/containers/ITunesProvider/AlbumDetails/index.js]: implemented the change

File: app/containers/ITunesProvider/saga.js (hunk 2 above)
Comment (bhaveshpawar07) [reply app/containers/ITunesProvider/saga.js]: implemented the change

**Review by naeemdadi1 — APPROVED** [review]

**Pull request merged**